crates/bevy_ui/src/render/ui_material_pipeline.rs

```diff
@@ -642,9 +642,11 @@ pub fn queue_ui_material_nodes<M: UiMaterial>(
                 bind_group_data: material.key.clone(),
             },
         );
-        transparent_phase
-            .items
-            .reserve(extracted_uinodes.uinodes.len());
+        if transparent_phase.items.capacity() < extracted_uinodes.uinodes.len() {
```

Comment:
The transparent phase items vec is shared between all  the extracted UI items, not just uimaterial nodes, so this isn't quite right either. Maybe the current length of items needs to be stored before the start of the loop, then this condition could become `transparent_phase.items.capacity() - initial_len < extracted.uinodes.len()`?

Reply:
so it seems we have no way to know the exact size required on each transparent phase, as not all nodes may go to the same if there are several cameras

Current behaviour on main is that for each entity, it's reserving "number of entities" capacity, so this PR is still an improvement

Reply:
Yeah that makes sense. We should merge this and look for further improvements later.